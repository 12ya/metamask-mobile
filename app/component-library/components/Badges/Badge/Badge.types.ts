// Internal dependencies.
import { BadgeAvatarProps } from './variants/BadgeAvatar/BadgeAvatar.types';

/**
 * Badge variants.
 */
export enum BadgeVariants {
  Avatar = 'avatar',
}

/**
 * Badge Account component props.
 */
export type BadgeProps = BadgeAvatarProps & {
  /**
   * Variant of badge.
   */
  variant: BadgeVariants;
};
