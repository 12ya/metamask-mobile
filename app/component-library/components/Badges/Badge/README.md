# Badge

Badge is a union component, which currently only consist of [BadgeNetwork](./variants/BadgeNetwork/BadgeNetwork.tsx)

## Common Props

### `variant`

Variant of badge.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> |
| :-------------------------------------------------- | :------------------------------------------------------ |
| [BadgeVariants](../../Badge.types.ts)                                           | Yes                                                     |

## BadgeNetwork Props

### `networkProps`

Props for the network content.

| <span style="color:gray;font-size:14px">TYPE</span> | <span style="color:gray;font-size:14px">REQUIRED</span> |
| :-------------------------------------------------- | :------------------------------------------------------ |
| [NetworkProps](../../Networks/Network/Network.types.ts)                                              | Yes                                                     |

## Usage

```javascript
const networkImageProps = {
    source: {
        uri: SAMPLE_IMAGE_URI
    }
}
const networkProps = {
  size: NetworkSizes.Md,
  name: SAMPLE_NETWORK_NAME,
  imageProps: networkImageProps,
}

<Badge
  variant={BadgeVariants.Network}
  networkProps={networkProps}
/>;
```