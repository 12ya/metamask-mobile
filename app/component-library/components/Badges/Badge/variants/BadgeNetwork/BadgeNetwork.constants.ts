/* eslint-disable import/prefer-default-export */
// Third party dependencies.
import { ImageSourcePropType } from 'react-native';

export const TEST_REMOTE_IMAGE_SOURCE: ImageSourcePropType = {
  uri: 'https://assets.coingecko.com/coins/images/279/small/ethereum.png?1595348880',
};
