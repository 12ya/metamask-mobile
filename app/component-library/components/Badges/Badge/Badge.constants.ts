/* eslint-disable import/prefer-default-export */

// External dependencies.
import { TEST_BADGE_NETWORK_PROPS } from './variants/BadgeNetwork/BadgeNetwork.constants';

// Internal dependencies.
import { BadgeProps, BadgeVariants } from './Badge.types';

// Test IDs
export const BADGE_NETWORK_TEST_ID = 'badge-network';

// Test consts
export const TEST_BADGE_PROPS: BadgeProps = {
  variant: BadgeVariants.Network,
  ...TEST_BADGE_NETWORK_PROPS,
};
