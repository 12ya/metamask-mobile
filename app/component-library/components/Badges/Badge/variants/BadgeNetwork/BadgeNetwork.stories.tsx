// Third party dependencies.
import React from 'react';

// External dependencies.
import { NetworkProps } from '../../../../Networks/Network/Network.types';
import { getNetworkStoryProps } from '../../../../Networks/Network/Network.stories';

// Internal dependencies.
import BadgeNetwork from './BadgeNetwork';
import { BadgeNetworkProps } from './BadgeNetwork.types';

export const getBadgeNetworkStoryProps = (): BadgeNetworkProps => {
  const networkProps: NetworkProps = getNetworkStoryProps();

  return {
    networkProps,
  };
};
const BadgeNetworkStory = () => (
  <BadgeNetwork {...getBadgeNetworkStoryProps()} />
);

export default BadgeNetworkStory;
