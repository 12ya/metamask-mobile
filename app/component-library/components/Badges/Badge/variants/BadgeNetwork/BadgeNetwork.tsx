/* eslint-disable react/prop-types */

// Third library dependencies.
import React from 'react';

// External dependencies.
import { useStyles } from '../../../../../hooks';
import BadgeBase from '../../foundation/BadgeBase';
import Network from '../../../../Networks/Network';

// Internal dependencies
import { BadgeNetworkProps } from './BadgeNetwork.types';
import styleSheet from './BadgeNetwork.styles';
import { BADGE_NETWORK_TEST_ID } from './BadgeNetwork.constants';

const BadgeNetwork = ({ networkProps, style }: BadgeNetworkProps) => {
  const { styles } = useStyles(styleSheet, { style });

  return (
    <BadgeBase style={styles.base} testID={BADGE_NETWORK_TEST_ID}>
      <Network {...networkProps} />
    </BadgeBase>
  );
};

export default BadgeNetwork;
