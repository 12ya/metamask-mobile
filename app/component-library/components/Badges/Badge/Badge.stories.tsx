/* eslint-disable no-console */
// Third party dependencies.
import React from 'react';
import { storiesOf } from '@storybook/react-native';

// External dependencies.
import { TEST_REMOTE_IMAGE_SOURCE } from './variants/BadgeNetwork/BadgeNetwork.constants';
import { BadgeVariants } from './Badge.types';

// Internal dependencies.
import Badge from './Badge';

storiesOf('Component Library / Badge', module).add('Default', () => (
  <Badge
    variant={BadgeVariants.Network}
    imageSource={TEST_REMOTE_IMAGE_SOURCE}
  />
));
