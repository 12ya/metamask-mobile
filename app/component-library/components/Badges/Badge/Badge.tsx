/* eslint-disable react/prop-types */
import React from 'react';

// External dependencies.
import BadgeAvatar from './variants/BadgeAvatar';

// Internal dependencies.
import { BadgeProps, BadgeVariants } from './Badge.types';
import { BADGE_AVATAR_TEST_ID } from './Badge.constants';

const Badge = (badgeProps: BadgeProps) => {
  switch (badgeProps.variant) {
    case BadgeVariants.Avatar:
      return <BadgeAvatar testID={BADGE_AVATAR_TEST_ID} {...badgeProps} />;
    default:
      throw new Error('Invalid Badge Variant');
  }
};

export default Badge;
