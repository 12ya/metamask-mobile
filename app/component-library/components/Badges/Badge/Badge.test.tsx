// Third party dependencies.
import React from 'react';
import { render, screen } from '@testing-library/react-native';

// External dependencies.
import { BadgeVariant } from './Badge.types';
import { SAMPLE_BADGENETWORK_PROPS } from './variants/BadgeNetwork/BadgeNetwork.constants';
import { SAMPLE_BADGESTATUS_PROPS } from './variants/BadgeStatus/BadgeStatus.constants';

// Internal dependencies.
import Badge from './Badge';
import {
  BADGE_BADGENETWORK_TEST_ID,
  BADGE_BADGESTATUS_TEST_ID,
  BADGE_BADGENOTIFICATIONS_TEST_ID,
} from './Badge.constants';

describe('Badge', () => {
  it('should render badge network given the badge network variant', () => {
    const { toJSON } = render(
      <Badge {...SAMPLE_BADGENETWORK_PROPS} variant={BadgeVariant.Network} />,
    );
    expect(toJSON()).toMatchSnapshot();
    const contentElement = screen.getByTestId(BADGE_BADGENETWORK_TEST_ID);
    expect(contentElement).toBeTruthy();
  });

  it('should render badge status given the badge status variant', () => {
    const { toJSON } = render(
      <Badge {...SAMPLE_BADGESTATUS_PROPS} variant={BadgeVariant.Status} />,
    );
    expect(toJSON()).toMatchSnapshot();
    const contentElement = screen.getByTestId(BADGE_BADGESTATUS_TEST_ID);
    expect(contentElement).toBeTruthy();
  });

  it('should render badge notifications given the badge notification variant', () => {
    const { toJSON } = render(
      <Badge
        {...SAMPLE_BADGESTATUS_PROPS}
        variant={BadgeVariant.NotificationsKinds}
      />,
    );
    expect(toJSON()).toMatchSnapshot();
    const contentElement = screen.getByTestId(BADGE_BADGENOTIFICATIONS_TEST_ID);
    expect(contentElement).toBeTruthy();
  });
});
