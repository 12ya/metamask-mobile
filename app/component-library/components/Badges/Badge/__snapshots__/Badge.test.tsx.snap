// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge - snapshots should render badge avatar given the badge avatar variant 1`] = `
<BadgeAvatar
  avatarProps={
    Object {
      "source": Object {
        "uri": "https://assets.coingecko.com/coins/images/279/small/ethereum.png?1595348880",
      },
      "variant": "Image",
    }
  }
  testID="badge-avatar"
  variant="avatar"
/>
`;
